deployer/debug.py

```diff
@@ -259,5 +264,128 @@ def start_docker_proxy(
         subprocess.check_call(cmd)
 
 
+@app.command()
+def transfer_old_home_dir_to_new_location(
+    cluster_name: str = typer.Argument(..., help="Name of cluster to operate on"),
+    hub_name: str = typer.Argument(..., help="Name of hub to operate on"),
+    source_user: str = typer.Option(
+        prompt=True,
+        confirmation_prompt=True,
+        help="Name of source user who's home dir files we are transferring",
+    ),
+    destination_user: str = typer.Option(
+        prompt=True,
+        confirmation_prompt=True,
+        help="Name of destination user in who's home dir we will be moving the files",
+    ),
+):
+    """
+    Copy all the files from a user's old home dir to the new home directory
+    """
+    # Name pod to include hub name so it is displayed as part of the default prompt
+    # This makes sure we don't end up deleting the wrong thing
+    pod_name = f"{cluster_name}-{hub_name}-transfer-shell"
+    pod = generate_pod_definition_with_home_mount(pod_name)
+
+    # Command that creates the pod described above
+    create_pod_cmd = [
+        "kubectl",
+        "-n",
+        hub_name,
+        "run",
+        pod_name,
+        "--overrides",
+        json.dumps(pod),
+        "--image",
+        # Use ubuntu image so we get GNU rm and other tools
+        # Should match what we have in our pod definition
+        "ubuntu:jammy",
+        pod_name,
+    ]
+
+    # Command that waits for the pod above to be ready
+    wait_for_pod_cmd = [
+        "kubectl",
+        "-n",
+        hub_name,
+        "wait",
+        "pods",
+        "-l",
+        f"run={pod_name}",
+        "--for",
+        "condition=Ready",
+        "--timeout=90s",
+    ]
+
+    # Command that lists the contents of the source and destination home dirs
+    ls_dirs_cmd = [
+        "kubectl",
+        "-n",
+        hub_name,
+        "exec",
+        pod_name,
+        "--",
+        "ls",
+        "-lat",
+        f"/home/{source_user}",
+        f"/home/{destination_user}",
+    ]
+
+    # Command that copies the home directory of the source user
+    # into the home directory of the destination user
+    # under a directory located at `/home/{destination_user}/{source_user}-homedir``
+    copy_cmd = [
+        "kubectl",
+        "-n",
+        hub_name,
+        "exec",
+        pod_name,
+        "--",
+        "cp",
+        "-r",
```

Comment:
Based on https://man7.org/linux/man-pages/man1/cp.1.html, I suggest adding the following options:

- `--archive`, to preserve permissions and stuff
- `--no-clobber`, so we don't accidentally delete files

Reply:
💯 yes!!!! Thank you!